src/modules/edc-demo/components/contract-agreement-transfer-dialog/contract-agreement-transfer-dialog-form.ts

```diff
@@ -1,65 +1,63 @@
 import {Injectable} from '@angular/core';
-import {FormBuilder, FormGroup, Validators} from '@angular/forms';
-import {switchDisabledControls} from '../../utils/form-group-utils';
-import {jsonValidator} from '../../validators/json-validator';
-import {urlValidator} from '../../validators/url-validator';
+import {FormBuilder, FormGroup} from '@angular/forms';
 import {DataAddressType} from '../data-address-type-select/data-address-type';
 import {
   ContractAgreementTransferDialogFormModel,
   ContractAgreementTransferDialogFormValue,
 } from './contract-agreement-transfer-dialog-form-model';
+import {ContractAgreementDatasinkFormBuilder} from './model/contract-agreement-datasink-form-builder';
+import {ContractAgreementDatasinkFormModel} from './model/contract-agreement-datasink-form-model';
 
 /**
  * Handles AngularForms for ContractAgreementTransferDialog
  */
 @Injectable()
 export class ContractAgreementTransferDialogForm {
-  formGroup = this.buildFormGroup();
+  all = this.buildFormGroup();
+
+  /**
+   * FormGroup for "Data Sink"
+   */
+  datasink = this.all.controls.datasink;
```

Comment:
It's fine if you want to rename "formGroup" to "all", since it's shorter, but this child form group should not exist, "all" is the datasink form.

Instead of accessing fields by "form.datasink.controls." you should use "form.all.controls."